File: src/app/dashboard/createmanager/page.js



"use client";
import { useState, useEffect } from "react";
import {
  Users, Mail, Lock, Building2, ShieldCheck, ArrowRight,
  Info, Loader2, Trash2, Search, UserPlus, Menu, X
} from "lucide-react";
import { toast } from "react-hot-toast";
import { apiFetch } from "../../../utils/apiClient";
import apiConfig from "../../../utils/apiConfig";
export default function ManagerCreatePage() {
  const [branches, setBranches] = useState([]);
  const [branchId, setBranchId] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [loading, setLoading] = useState(true);
  const [submitting, setSubmitting] = useState(false);
  const [searchTerm, setSearchTerm] = useState("");
  const [permissionLevel, setPermissionLevel] = useState("LIMITED");

  // Mobile Sidebar Toggle
  const [isSidebarOpen, setIsSidebarOpen] = useState(false);

  const getToken = () => typeof window !== "undefined" ? localStorage.getItem("token") : null;




  const loadBranches = async () => {
    try {
      const res = await fetch(`${apiConfig?.BASE_URL}/api/restaurants`, {
        credentials: "include"   // 🔥 CRITICAL
      });

      const data = await res.json();

      if (!res.ok) {
        throw new Error(data.message || "Failed to load branches");
      }

      setBranches(data);

    } catch (err) {
      toast.error(err.message || "Network error");
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    loadBranches();
  }, []);



  const submit = async () => {
    if (!branchId || !email || !password || !permissionLevel)
      return toast.error("Please fill all fields");

    setSubmitting(true);

    console.log(email,
      password,
      permissionLevel, "Email , Password", "permissionLevel");

    try {
      const res = await fetch(
        `${apiConfig?.BASE_URL}/api/restaurants/${branchId}/assign-manager`,
        {
          method: "POST",
          credentials: "include",   // 🔥 CRITICAL
          headers: {
            "Content-Type": "application/json"
          },
          body: JSON.stringify({
            email,
            password,
            permissionLevel
          })
        }
      );

      const data = await res.json();

      if (!res.ok) {
        throw new Error(data.message || "Assignment failed");
      }

      toast.success("Manager assigned successfully!");
      setEmail("");
      setPassword("");
      setIsSidebarOpen(false);

    } catch (err) {
      toast.error(err.message || "Connection failed");
    } finally {
      setSubmitting(false);
    }
  };


  if (loading) {
    return (
      <div className="flex items-center justify-center h-screen bg-[#FDFCF6]">
        <div className="animate-spin rounded-full h-12 w-12 border-t-4 border-[#FF4D00]"></div>
      </div>
    );
  }
  return (
    <div className="h-screen bg-[#F8FAFC] flex flex-col overflow-hidden">
      {/* 🚀 Header Action Bar */}
      <div className="bg-white border-b border-slate-200 px-4 md:px-8 py-4 md:py-5 flex items-center justify-between shrink-0">
        <div>
          <h1 className="text-xl md:text-2xl font-black text-slate-900 tracking-tight">Staff</h1>
          <p className="hidden md:block text-slate-500 text-xs font-medium uppercase tracking-wider">Access Control</p>
        </div>

        <div className="flex items-center gap-2 md:gap-4">
          <div className="relative hidden sm:block">
            <Search className="absolute left-3 top-1/2 -translate-y-1/2 text-slate-400" size={16} />
            <input
              type="text"
              placeholder="Search..."
              className="pl-10 pr-4 py-2 bg-slate-100 rounded-lg text-sm outline-none w-40 md:w-64 focus:ring-2 focus:ring-orange-500/20 transition-all"
              value={searchTerm}
              onChange={(e) => setSearchTerm(e.target.value)}
            />
          </div>

          {/* Mobile Toggle Button */}
          <button
            onClick={() => setIsSidebarOpen(true)}
            className="lg:hidden bg-orange-500 text-white p-2 rounded-lg flex items-center gap-2 px-3 text-xs font-bold"
          >
            <UserPlus size={16} /> <span className="hidden xs:block">New</span>
          </button>
        </div>
      </div>

      <div className="flex flex-1 overflow-hidden relative">
        {/* 📝 Left Pane: Overlay & Registration Form */}
        {/* Mobile Backdrop */}
        {isSidebarOpen && (
          <div
            className="fixed inset-0 bg-slate-900/40 z-40 lg:hidden backdrop-blur-sm"
            onClick={() => setIsSidebarOpen(false)}
          />
        )}

        <aside className={`
          fixed inset-y-0 left-0 z-50 w-[85%] sm:w-[400px] bg-white p-6 md:p-8 overflow-y-auto shadow-2xl transition-transform duration-300 transform
          lg:static lg:translate-x-0 lg:w-[450px] lg:shadow-none lg:border-r lg:border-slate-200
          ${isSidebarOpen ? 'translate-x-0' : '-translate-x-full'}
        `}>
          <div className="flex items-center justify-between mb-6 lg:mb-8">
            <div className="inline-flex items-center gap-2 bg-orange-50 text-orange-600 px-3 py-1 rounded-full">
              <UserPlus size={14} />
              <span className="text-[10px] font-black uppercase tracking-widest">New Assignment</span>
            </div>
            <button onClick={() => setIsSidebarOpen(false)} className="lg:hidden p-2 text-slate-400">
              <X size={20} />
            </button>
          </div>

          <div className="space-y-5 md:space-y-6">
            <div>
              <label className="text-[10px] font-black text-slate-400 uppercase tracking-widest mb-2 block">Target Branch</label>
              <div className="relative group">
                <Building2 className="absolute left-4 top-1/2 -translate-y-1/2 text-slate-400" size={18} />
                <select
                  disabled={loading}
                  value={branchId}
                  onChange={e => setBranchId(e.target.value)}
                  className="w-full pl-12 pr-4 py-3 md:py-4 rounded-xl bg-slate-50 border border-slate-200 focus:border-orange-500 outline-none font-bold text-slate-700 appearance-none"
                >
                  <option value="">{loading ? "Loading..." : "Select Outlet"}</option>
                  {branches.map(b => <option key={b._id} value={b._id}>{b.name}</option>)}
                </select>
              </div>
            </div>

            <div>
              <label className="text-[10px] font-black text-slate-400 uppercase tracking-widest mb-2 block">Login Email</label>
              <div className="relative group">
                <Mail className="absolute left-4 top-1/2 -translate-y-1/2 text-slate-400" size={18} />
                <input
                  value={email}
                  placeholder="manager@email.com"
                  className="w-full pl-12 pr-4 py-3 md:py-4 rounded-xl bg-slate-50 border border-slate-200 focus:border-orange-500 outline-none font-bold text-slate-700"
                  onChange={e => setEmail(e.target.value)}
                />
              </div>
            </div>

            <div>
              <label className="text-[10px] font-black text-slate-400 uppercase tracking-widest mb-2 block">Access Password</label>
              <div className="relative group">
                <Lock className="absolute left-4 top-1/2 -translate-y-1/2 text-slate-400" size={18} />
                <input
                  type="password"
                  value={password}
                  placeholder="••••••••"
                  className="w-full pl-12 pr-4 py-3 md:py-4 rounded-xl bg-slate-50 border border-slate-200 focus:border-orange-500 outline-none font-bold text-slate-700"
                  onChange={e => setPassword(e.target.value)}
                />
              </div>
            </div>

            <div>
              <label className="text-[10px] font-black text-slate-400 uppercase tracking-widest mb-2 block">
                Manager Access Level
              </label>

              <div className="flex gap-3">
                <button
                  type="button"
                  onClick={() => setPermissionLevel("LIMITED")}
                  className={`flex-1 py-3 rounded-xl border font-bold text-xs uppercase tracking-wider transition-all ${permissionLevel === "LIMITED"
                    ? "bg-slate-900 text-white border-slate-900"
                    : "bg-slate-50 text-slate-600 border-slate-200"
                    }`}
                >
                  Operations Only
                </button>

                <button
                  type="button"
                  onClick={() => setPermissionLevel("FULL")}
                  className={`flex-1 py-3 rounded-xl border font-bold text-xs uppercase tracking-wider transition-all ${permissionLevel === "FULL"
                    ? "bg-orange-600 text-white border-orange-600"
                    : "bg-slate-50 text-slate-600 border-slate-200"
                    }`}
                >
                  Full Financial Access
                </button>
              </div>
            </div>



            <button
              disabled={submitting}
              onClick={submit}
              className="w-full bg-slate-900 text-white py-4 md:py-5 rounded-2xl font-black text-[11px] uppercase tracking-[0.2em] hover:bg-orange-600 transition-all flex items-center justify-center gap-3 active:scale-95 shadow-lg"
            >
              {submitting ? <Loader2 className="animate-spin" size={18} /> : <>Deploy Manager <ArrowRight size={16} /></>}
            </button>
          </div>
        </aside>

        {/* 📋 Right Pane: Live Staff Ledger */}
        <main className="flex-1 p-4 md:p-8 overflow-y-auto bg-[#F8FAFC]">
          <div className="flex flex-col sm:flex-row sm:items-center justify-between mb-6 md:mb-8 gap-4">
            <h2 className="text-sm font-black text-slate-800 uppercase tracking-widest flex items-center gap-2">
              <Users size={18} className="text-slate-400" /> Management Team
            </h2>
            <div className="flex items-center gap-2 text-[10px] font-bold text-slate-400 uppercase">
              <ShieldCheck size={14} className="text-green-500" /> Secure Cloud Sync
            </div>
          </div>

          <div className="grid grid-cols-1 md:grid-cols-1 xl:grid-cols-2 gap-4">
            {branches.map(branch => (
              <div key={branch._id} className="bg-white p-5 md:p-6 rounded-[1.5rem] md:rounded-[2rem] border border-slate-200 hover:border-orange-200 transition-all">
                <div className="flex justify-between items-start mb-4">
                  <div className="w-10 h-10 md:w-12 md:h-12 bg-slate-100 rounded-xl md:rounded-2xl flex items-center justify-center text-slate-500">
                    <Building2 size={20} md={24} />
                  </div>
                  <button className="p-2 text-slate-300 hover:text-red-500 transition-colors">
                    <Trash2 size={16} />
                  </button>
                </div>
                <div>
                  <h3 className="font-black text-slate-900 text-base md:text-lg mb-1">{branch.name}</h3>
                  <div className="flex items-center gap-2 text-slate-500 mb-4">
                    <Mail size={12} />
                    <span className="text-xs font-bold truncate max-w-[200px]">{branch?.manager?.email || "No manager assigned"}</span>
                  </div>
                  <div className="flex items-center justify-between pt-4 border-t border-slate-50">
                    <span className="text-[9px] font-black uppercase tracking-widest text-slate-400">Status</span>
                    <span className={`px-2 py-0.5 rounded text-[9px] font-black uppercase ${branch?.manager ? 'bg-emerald-100 text-emerald-700' : 'bg-slate-100 text-slate-500'}`}>
                      {branch?.manager ? 'Active' : 'Vacant'}
                    </span>
                  </div>
                </div>
              </div>
            ))}
          </div>
        </main>
      </div>
    </div>
  );
}

File: lib/src/app/dashboard/createmanager/page.dart
import 'package:flutter/material.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';
import 'package:lucide_icons/lucide_icons.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../../utils/apiClient.dart';

class ManagerCreatePage extends ConsumerStatefulWidget {
  const ManagerCreatePage({super.key});
  @override
  ConsumerState<ManagerCreatePage> createState() => _ManagerCreatePageState();
}

class _ManagerCreatePageState extends ConsumerState<ManagerCreatePage> {
  List<dynamic> _branches = [];
  String _branchId = '';
  String _email = '';
  String _password = '';
  String _permissionLevel = 'LIMITED';
  String _searchTerm = '';

  bool _isLoading = true;
  bool _isSubmitting = false;

  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();

  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((_) {
      _loadBranches();
    });
  }

  Future<void> _loadBranches() async {
    setState(() => _isLoading = true);
    try {
      final res = await apiFetch('/api/restaurants');
      if (res is List) {
        if (mounted) setState(() => _branches = res);
      } else {
        throw Exception(res['message'] ?? 'Failed to load branches');
      }
    } catch (e) {
      if (mounted)
        ScaffoldMessenger.of(
          context,
        ).showSnackBar(SnackBar(content: Text('Network error $e')));
    } finally {
      if (mounted) setState(() => _isLoading = false);
    }
  }

  Future<void> _submit() async {
    if (_branchId.isEmpty ||
        _email.isEmpty ||
        _password.isEmpty ||
        _permissionLevel.isEmpty) {
      ScaffoldMessenger.of(
        context,
      ).showSnackBar(const SnackBar(content: Text('Please fill all fields')));
      return;
    }

    setState(() => _isSubmitting = true);
    try {
      final res = await apiFetch(
        '/api/restaurants/$_branchId/assign-manager',
        method: 'POST',
        data: {
          'email': _email,
          'password': _password,
          'permissionLevel': _permissionLevel,
        },
      );

      if (res['manager'] != null ||
          res['message']?.toString().toLowerCase().contains('success') ==
              true) {
        if (mounted) {
          ScaffoldMessenger.of(context).showSnackBar(
            const SnackBar(content: Text('Manager assigned successfully!')),
          );
          setState(() {
            _email = '';
            _password = '';
          });
          _loadBranches();
          if (_scaffoldKey.currentState?.isDrawerOpen ?? false) {
            Navigator.pop(context);
          }
        }
      } else {
        throw Exception(res['message'] ?? 'Assignment failed');
      }
    } catch (e) {
      if (mounted)
        ScaffoldMessenger.of(
          context,
        ).showSnackBar(SnackBar(content: Text('Connection failed: $e')));
    } finally {
      if (mounted) setState(() => _isSubmitting = false);
    }
  }

  Future<void> _removeManager(String branchId) async {
    final act = await showDialog<bool>(
      context: context,
      builder: (ctx) => AlertDialog(
        title: const Text('Revoke Manager Access?'),
        content: const Text(
          'Are you sure you want to remove the manager for this branch? This action cannot be undone.',
        ),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(ctx, false),
            child: const Text('CANCEL'),
          ),
          TextButton(
            onPressed: () => Navigator.pop(ctx, true),
            child: const Text(
              'REMOVE',
              style: const TextStyle(color: Colors.red),
            ),
          ),
        ],
      ),
    );
    if (act != true) return;

    try {
      await apiFetch(
        '/api/restaurants/$branchId/remove-manager',
        method: 'DELETE',
      );
      await _loadBranches();
      if (mounted)
        ScaffoldMessenger.of(context).showSnackBar(
          const SnackBar(content: Text('Manager access revoked successfully')),
        );
    } catch (e) {
      if (mounted)
        ScaffoldMessenger.of(
          context,
        ).showSnackBar(SnackBar(content: Text('Failed to revoke manager: $e')));
    }
  }

  @override
  Widget build(BuildContext context) {
    if (_isLoading) {
      return const Scaffold(
        backgroundColor: Color(0xFFFDFCF6),
        body: Center(
          child: CircularProgressIndicator(color: Color(0xFFFF4D00)),
        ),
      );
    }

    final filteredBranches = _branches.where((b) {
      final name = (b['name'] ?? '').toString().toLowerCase();
      return name.contains(_searchTerm.toLowerCase());
    }).toList();

    return Scaffold(
      key: _scaffoldKey,
      backgroundColor: const Color(0xFFF8FAFC),
      drawer: _buildDrawer(),
      body: Column(
        children: [
          // Header Action Bar
          Container(
            padding: EdgeInsets.symmetric(horizontal: 24.w, vertical: 20.h),
            decoration: const BoxDecoration(
              color: Colors.white,
              border: Border(bottom: BorderSide(color: Colors.black12)),
            ),
            child: Wrap(
              alignment: WrapAlignment.spaceBetween,
              crossAxisAlignment: WrapCrossAlignment.center,
              spacing: 16.w,
              runSpacing: 16.h,
              children: [
                Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      'Staff',
                      style: TextStyle(
                        fontSize: 24.sp,
                        fontWeight: FontWeight.w900,
                        color: Color(0xFF0F172A),
                        letterSpacing: -1,
                      ),
                    ),
                    Text(
                      'ACCESS CONTROL',
                      style: TextStyle(
                        fontSize: 10.sp,
                        fontWeight: FontWeight.w900,
                        color: Colors.grey,
                        letterSpacing: 1,
                      ),
                    ),
                  ],
                ),
                LayoutBuilder(
                  builder: (ctx, constraints) {
                    final isDesktop = MediaQuery.of(context).size.width > 900.w;
                    return Row(
                      mainAxisSize: MainAxisSize.min,
                      children: [
                        Container(
                          width: isDesktop ? 250.w : 200.w,
                          height: 48.h,
                          padding: EdgeInsets.symmetric(horizontal: 16.w),
                          decoration: BoxDecoration(
                            color: Colors.grey.shade50,
                            borderRadius: BorderRadius.circular(16.r),
                          ),
                          child: Row(
                            children: [
                              Icon(
                                LucideIcons.search,
                                size: 16.sp,
                                color: Colors.grey,
                              ),
                              SizedBox(width: 12.w),
                              Expanded(
                                child: TextField(
                                  onChanged: (v) =>
                                      setState(() => _searchTerm = v),
                                  decoration: const InputDecoration(
                                    hintText: 'Search...',
                                    border: InputBorder.none,
                                  ),
                                  style: TextStyle(
                                    fontSize: 14.sp,
                                    fontWeight: FontWeight.bold,
                                  ),
                                ),
                              ),
                            ],
                          ),
                        ),
                        if (!isDesktop) ...[
                          SizedBox(width: 16.w),
                          ElevatedButton.icon(
                            onPressed: () =>
                                _scaffoldKey.currentState?.openDrawer(),
                            icon: Icon(LucideIcons.userPlus, size: 16.sp),
                            label: Text(
                              'NEW',
                              style: TextStyle(fontWeight: FontWeight.w900),
                            ),
                            style: ElevatedButton.styleFrom(
                              backgroundColor: const Color(0xFFFF4D00),
                              foregroundColor: Colors.white,
                              padding: EdgeInsets.symmetric(
                                horizontal: 16.w,
                                vertical: 16.h,
                              ),
                              shape: RoundedRectangleBorder(
                                borderRadius: BorderRadius.circular(12.r),
                              ),
                            ),
                          ),
                        ],
                      ],
                    );
                  },
                ),
              ],
            ),
          ),

          Expanded(
            child: Row(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                // Left Pane: Registration Form (Desktop)
                LayoutBuilder(
                  builder: (ctx, constraints) {
                    if (MediaQuery.of(context).size.width <= 900.w)
                      return const SizedBox.shrink();
                    return _buildFormPane();
                  },
                ),

                // Right Pane: Live Staff Ledger
                Expanded(
                  child: Padding(
                    padding: EdgeInsets.all(32.r),
                    child: Column(
                      children: [
                        Row(
                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
                          children: [
                            Row(
                              children: [
                                Icon(
                                  LucideIcons.users,
                                  size: 18.sp,
                                  color: Colors.grey,
                                ),
                                SizedBox(width: 8.w),
                                Text(
                                  'MANAGEMENT TEAM',
                                  style: TextStyle(
                                    fontSize: 12.sp,
                                    fontWeight: FontWeight.w900,
                                    color: Color(0xFF0F172A),
                                    letterSpacing: 1,
                                  ),
                                ),
                              ],
                            ),
                            Row(
                              children: [
                                Icon(
                                  LucideIcons.shieldCheck,
                                  size: 14.sp,
                                  color: Colors.green,
                                ),
                                SizedBox(width: 8.w),
                                Text(
                                  'SECURE CLOUD SYNC',
                                  style: TextStyle(
                                    fontSize: 10.sp,
                                    fontWeight: FontWeight.bold,
                                    color: Colors.grey,
                                    letterSpacing: 1,
                                  ),
                                ),
                              ],
                            ),
                          ],
                        ),
                        SizedBox(height: 32.h),

                        Expanded(
                          child: GridView.builder(
                            gridDelegate:
                                SliverGridDelegateWithMaxCrossAxisExtent(
                                  maxCrossAxisExtent: 400.w,
                                  mainAxisExtent: 220.h,
                                  crossAxisSpacing: 16.w,
                                  mainAxisSpacing: 16.h,
                                ),
                            itemCount: filteredBranches.length,
                            itemBuilder: (ctx, idx) {
                              final b = filteredBranches[idx];
                              final hasManager = b['manager'] != null;
                              return Container(
                                padding: EdgeInsets.all(24.r),
                                decoration: BoxDecoration(
                                  color: Colors.white,
                                  borderRadius: BorderRadius.circular(32.r),
                                  border: Border.all(
                                    color: Colors.grey.shade200,
                                  ),
                                ),
                                child: Column(
                                  crossAxisAlignment: CrossAxisAlignment.start,
                                  children: [
                                    Row(
                                      mainAxisAlignment:
                                          MainAxisAlignment.spaceBetween,
                                      children: [
                                        Container(
                                          width: 48.w,
                                          height: 48.h,
                                          decoration: BoxDecoration(
                                            color: Colors.grey.shade50,
                                            borderRadius: BorderRadius.circular(
                                              16.r,
                                            ),
                                          ),
                                          child: Icon(
                                            LucideIcons.building2,
                                            size: 24.sp,
                                            color: Colors.grey,
                                          ),
                                        ),
                                        if (hasManager)
                                          IconButton(
                                            onPressed: () =>
                                                _removeManager(b['_id']),
                                            icon: Icon(
                                              LucideIcons.trash2,
                                              size: 16.sp,
                                              color: Colors.red,
                                            ),
                                          )
                                        else
                                          IconButton(
                                            onPressed: () {},
                                            icon: Icon(
                                              LucideIcons.trash2,
                                              size: 16.sp,
                                              color: Colors.grey,
                                            ),
                                          ),
                                      ],
                                    ),
                                    const Spacer(),
                                    Text(
                                      b['name'] ?? 'Unnamed',
                                      style: TextStyle(
                                        fontSize: 18.sp,
                                        fontWeight: FontWeight.w900,
                                        color: Color(0xFF0F172A),
                                      ),
                                    ),
                                    SizedBox(height: 8.h),
                                    Row(
                                      children: [
                                        Icon(
                                          LucideIcons.mail,
                                          size: 12.sp,
                                          color: Colors.grey,
                                        ),
                                        SizedBox(width: 8.w),
                                        Expanded(
                                          child: Text(
                                            b['manager']?['email'] ??
                                                'No manager assigned',
                                            style: TextStyle(
                                              fontSize: 12.sp,
                                              fontWeight: FontWeight.bold,
                                              color: Colors.grey,
                                            ),
                                            maxLines: 1,
                                          ),
                                        ),
                                      ],
                                    ),
                                    SizedBox(height: 16.h),
                                    Container(
                                      padding: EdgeInsets.symmetric(
                                        vertical: 12.h,
                                      ),
                                      decoration: BoxDecoration(
                                        border: Border(
                                          top: BorderSide(
                                            color: Colors.black12,
                                          ),
                                        ),
                                      ),
                                      child: Row(
                                        mainAxisAlignment:
                                            MainAxisAlignment.spaceBetween,
                                        children: [
                                          Text(
                                            'STATUS',
                                            style: TextStyle(
                                              fontSize: 9.sp,
                                              fontWeight: FontWeight.w900,
                                              color: Colors.grey,
                                              letterSpacing: 2,
                                            ),
                                          ),
                                          Container(
                                            padding: EdgeInsets.symmetric(
                                              horizontal: 8.w,
                                              vertical: 4.h,
                                            ),
                                            decoration: BoxDecoration(
                                              color: hasManager
                                                  ? Colors.green.shade50
                                                  : Colors.grey.shade100,
                                              borderRadius:
                                                  BorderRadius.circular(4.r),
                                            ),
                                            child: Text(
                                              hasManager ? 'ACTIVE' : 'VACANT',
                                              style: TextStyle(
                                                fontSize: 9.sp,
                                                fontWeight: FontWeight.w900,
                                                color: hasManager
                                                    ? Colors.green.shade700
                                                    : Colors.grey,
                                              ),
                                            ),
                                          ),
                                        ],
                                      ),
                                    ),
                                  ],
                                ),
                              );
                            },
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildDrawer() {
    return Drawer(
      backgroundColor: Colors.white,
      width: MediaQuery.of(context).size.width > 400.w
          ? 400.w
          : MediaQuery.of(context).size.width * 0.9,
      child: SafeArea(child: _buildFormPane(isDrawer: true)),
    );
  }

  Widget _buildFormPane({bool isDrawer = false}) {
    return Container(
      width: isDrawer ? double.infinity : 450.w,
      padding: EdgeInsets.all(isDrawer ? 24.r : 32.r),
      decoration: BoxDecoration(
        color: Colors.white,
        border: Border(right: BorderSide(color: Colors.grey.shade200)),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Container(
            padding: EdgeInsets.symmetric(horizontal: 12.w, vertical: 6.h),
            decoration: BoxDecoration(
              color: Colors.orange.shade50,
              borderRadius: BorderRadius.circular(20.r),
            ),
            child: Row(
              mainAxisSize: MainAxisSize.min,
              children: [
                Icon(LucideIcons.userPlus, size: 14.sp, color: Colors.orange),
                SizedBox(width: 8.w),
                Text(
                  'NEW ASSIGNMENT',
                  style: TextStyle(
                    fontSize: 10.sp,
                    fontWeight: FontWeight.w900,
                    color: Colors.orange,
                    letterSpacing: 1,
                  ),
                ),
              ],
            ),
          ),
          SizedBox(height: 32.h),

          Expanded(
            child: ListView(
              children: [
                Text(
                  'TARGET BRANCH',
                  style: TextStyle(
                    fontSize: 10.sp,
                    fontWeight: FontWeight.w900,
                    color: Colors.grey,
                    letterSpacing: 1,
                  ),
                ),
                SizedBox(height: 8.h),
                Container(
                  padding: EdgeInsets.symmetric(horizontal: 16.w),
                  decoration: BoxDecoration(
                    color: Colors.grey.shade50,
                    borderRadius: BorderRadius.circular(16.r),
                    border: Border.all(color: Colors.grey.shade200),
                  ),
                  child: Row(
                    children: [
                      Icon(
                        LucideIcons.building2,
                        size: 18.sp,
                        color: Colors.grey,
                      ),
                      SizedBox(width: 12.w),
                      Expanded(
                        child: DropdownButtonHideUnderline(
                          child: DropdownButton<String>(
                            isExpanded: true,
                            hint: Text(
                              'Select Outlet',
                              style: TextStyle(
                                fontSize: 14.sp,
                                fontWeight: FontWeight.bold,
                                color: Colors.grey,
                              ),
                            ),
                            value: _branchId.isEmpty ? null : _branchId,
                            items: _branches
                                .map(
                                  (b) => DropdownMenuItem<String>(
                                    value: b['_id'],
                                    child: Text(
                                      b['name'],
                                      style: TextStyle(
                                        fontSize: 14.sp,
                                        fontWeight: FontWeight.bold,
                                      ),
                                    ),
                                  ),
                                )
                                .toList(),
                            onChanged: (v) {
                              if (v != null) setState(() => _branchId = v);
                            },
                          ),
                        ),
                      ),
                    ],
                  ),
                ),
                SizedBox(height: 24.h),

                Text(
                  'LOGIN EMAIL',
                  style: TextStyle(
                    fontSize: 10.sp,
                    fontWeight: FontWeight.w900,
                    color: Colors.grey,
                    letterSpacing: 1,
                  ),
                ),
                SizedBox(height: 8.h),
                Container(
                  padding: EdgeInsets.symmetric(
                    horizontal: 16.w,
                    vertical: 4.h,
                  ),
                  decoration: BoxDecoration(
                    color: Colors.grey.shade50,
                    borderRadius: BorderRadius.circular(16.r),
                    border: Border.all(color: Colors.grey.shade200),
                  ),
                  child: Row(
                    children: [
                      Icon(LucideIcons.mail, size: 18.sp, color: Colors.grey),
                      SizedBox(width: 12.w),
                      Expanded(
                        child: TextField(
                          onChanged: (v) => _email = v,
                          decoration: InputDecoration(
                            hintText: 'manager@email.com',
                            border: InputBorder.none,
                          ),
                          style: TextStyle(
                            fontSize: 14.sp,
                            fontWeight: FontWeight.bold,
                          ),
                        ),
                      ),
                    ],
                  ),
                ),
                SizedBox(height: 24.h),

                Text(
                  'ACCESS PASSWORD',
                  style: TextStyle(
                    fontSize: 10.sp,
                    fontWeight: FontWeight.w900,
                    color: Colors.grey,
                    letterSpacing: 1,
                  ),
                ),
                SizedBox(height: 8.h),
                Container(
                  padding: EdgeInsets.symmetric(
                    horizontal: 16.w,
                    vertical: 4.h,
                  ),
                  decoration: BoxDecoration(
                    color: Colors.grey.shade50,
                    borderRadius: BorderRadius.circular(16.r),
                    border: Border.all(color: Colors.grey.shade200),
                  ),
                  child: Row(
                    children: [
                      Icon(LucideIcons.lock, size: 18.sp, color: Colors.grey),
                      SizedBox(width: 12.w),
                      Expanded(
                        child: TextField(
                          obscureText: true,
                          onChanged: (v) => _password = v,
                          decoration: InputDecoration(
                            hintText: '••••••••',
                            border: InputBorder.none,
                          ),
                          style: TextStyle(
                            fontSize: 14.sp,
                            fontWeight: FontWeight.bold,
                          ),
                        ),
                      ),
                    ],
                  ),
                ),
                SizedBox(height: 24.h),

                Text(
                  'MANAGER ACCESS LEVEL',
                  style: TextStyle(
                    fontSize: 10.sp,
                    fontWeight: FontWeight.w900,
                    color: Colors.grey,
                    letterSpacing: 1,
                  ),
                ),
                SizedBox(height: 8.h),
                Row(
                  children: [
                    Expanded(
                      child: GestureDetector(
                        onTap: () =>
                            setState(() => _permissionLevel = 'LIMITED'),
                        child: Container(
                          padding: EdgeInsets.symmetric(vertical: 16.h),
                          decoration: BoxDecoration(
                            color: _permissionLevel == 'LIMITED'
                                ? const Color(0xFF0F172A)
                                : Colors.grey.shade50,
                            borderRadius: BorderRadius.circular(16.r),
                            border: Border.all(
                              color: _permissionLevel == 'LIMITED'
                                  ? const Color(0xFF0F172A)
                                  : Colors.grey.shade200,
                            ),
                          ),
                          alignment: Alignment.center,
                          child: Text(
                            'OPERATIONS ONLY',
                            style: TextStyle(
                              fontSize: 10.sp,
                              fontWeight: FontWeight.w900,
                              color: _permissionLevel == 'LIMITED'
                                  ? Colors.white
                                  : Colors.grey.shade600,
                              letterSpacing: 1,
                            ),
                          ),
                        ),
                      ),
                    ),
                    SizedBox(width: 12.w),
                    Expanded(
                      child: GestureDetector(
                        onTap: () => setState(() => _permissionLevel = 'FULL'),
                        child: Container(
                          padding: EdgeInsets.symmetric(vertical: 16.h),
                          decoration: BoxDecoration(
                            color: _permissionLevel == 'FULL'
                                ? const Color(0xFFFF4D00)
                                : Colors.grey.shade50,
                            borderRadius: BorderRadius.circular(16.r),
                            border: Border.all(
                              color: _permissionLevel == 'FULL'
                                  ? const Color(0xFFFF4D00)
                                  : Colors.grey.shade200,
                            ),
                          ),
                          alignment: Alignment.center,
                          child: Text(
                            'FULL FINANCIAL',
                            style: TextStyle(
                              fontSize: 10.sp,
                              fontWeight: FontWeight.w900,
                              color: _permissionLevel == 'FULL'
                                  ? Colors.white
                                  : Colors.grey.shade600,
                              letterSpacing: 1,
                            ),
                          ),
                        ),
                      ),
                    ),
                  ],
                ),
                SizedBox(height: 32.h),

                ElevatedButton.icon(
                  onPressed: _isSubmitting ? null : _submit,
                  icon: Icon(LucideIcons.arrowRight, size: 16.sp),
                  label: Text(
                    _isSubmitting ? 'WORKING...' : 'DEPLOY MANAGER',
                    style: TextStyle(
                      fontSize: 11.sp,
                      fontWeight: FontWeight.w900,
                      letterSpacing: 2,
                    ),
                  ),
                  style: ElevatedButton.styleFrom(
                    backgroundColor: const Color(0xFF0F172A),
                    foregroundColor: Colors.white,
                    padding: EdgeInsets.symmetric(vertical: 24.h),
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(16.r),
                    ),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}
